pkg/controllers/machinemigration/machine_migration_controller.go

```diff
@@ -0,0 +1,492 @@
+		if err != nil {
+			return false, fmt.Errorf("unable to get paused condition for %s/%s: %w", m.Namespace, m.Name, err)
+		}
+
+		return cond == corev1.ConditionTrue, nil
+	}
+
+	// For MachineAuthorityMachineAPI, check the corresponding CAPI resource.
+	capiMachine := &capiv1beta1.Machine{}
+	if err := r.Get(ctx, client.ObjectKey{Namespace: consts.DefaultManagedNamespace, Name: m.Name}, capiMachine); err != nil {
+		return false, fmt.Errorf("failed to get Cluster API machine: %w", err)
+	}
+
+	machinePausedCondition := conditionsv1beta2.Get(capiMachine, capiv1beta1.PausedV1Beta2Condition)
+	if machinePausedCondition == nil {
+		return false, nil
+	}
+
+	infraMachineRef := capiMachine.Spec.InfrastructureRef
+
+	infraMachine, err := util.GetReferencedObject(ctx, r.Client, r.Scheme, infraMachineRef)
+	if err != nil {
+		return false, fmt.Errorf("failed to get Cluster API infra machine: %w", err)
+	}
+
+	infraMachinePausedConditionStatus, err := util.GetConditionStatus(infraMachine, capiv1beta1.PausedV1Beta2Condition)
+	if err != nil {
+		return false, fmt.Errorf("unable to get paused condition for %s/%s: %w", infraMachine.GetNamespace(), infraMachine.GetName(), err)
+	}
+
+	return (machinePausedCondition.Status == metav1.ConditionTrue) && (infraMachinePausedConditionStatus == corev1.ConditionTrue), nil
+}
+
+func (r *MachineMigrationReconciler) ensureUnpauseRequestedOnNewAuthoritativeResource(ctx context.Context, mapiMachine *machinev1beta1.Machine) error {
+	// Request that the new authoritative resource reconciliation is un-paused.
+	//nolint:wsl,exhaustive
+	switch mapiMachine.Spec.AuthoritativeAPI {
+	case machinev1beta1.MachineAuthorityClusterAPI:
+		// For requesting unpausing of a CAPI resource, remove the paused annotation on it.
+		// So check if the ClusterAPI resource has the paused annotation and if so remove it.
+		capiMachine := &capiv1beta1.Machine{}
+		if err := r.Get(ctx, client.ObjectKey{Namespace: consts.DefaultManagedNamespace, Name: mapiMachine.Name}, capiMachine); err != nil {
+			return fmt.Errorf("failed to get Cluster API machine: %w", err)
+		}
+
+		infraMachineRef := capiMachine.Spec.InfrastructureRef
+
+		infraMachine, err := util.GetReferencedObject(ctx, r.Client, r.Scheme, infraMachineRef)
+		if err != nil {
+			return fmt.Errorf("failed to get Cluster API infra machine: %w", err)
+		}
+
+		if annotations.HasPaused(capiMachine) {
+			capiMachineCopy := capiMachine.DeepCopy()
+			delete(capiMachine.Annotations, capiv1beta1.PausedAnnotation)
+
+			if err := r.Patch(ctx, capiMachine, client.MergeFrom(capiMachineCopy)); err != nil {
+				return fmt.Errorf("failed to patch Cluster API machine: %w", err)
+			}
+		}
+
+		if annotations.HasPaused(infraMachine) {
+			infraMachineCopy, ok := infraMachine.DeepCopyObject().(client.Object)
+			if !ok {
+				return fmt.Errorf("unable to assert Cluster API infra machine as client.Object: %w", err)
+			}
+			util.RemoveAnnotation(infraMachine, capiv1beta1.PausedAnnotation)
```

Comment:
Why util here when delete is used on 261? What's the difference?

Reply:
`delete` only works on maps and their keys.

the `infraMachine` is a client.Object and we don't have direct access to its map, so the util gets it, does the delete and sets back.

Reply:
`client.Object` has `GetAnnotations()` doesn't it? So `delete(infraMachine.GetAnnotations(), capiv1beta1.PausedAnnotation)` should work here